### ISSUE TITLE
Moderation strategies against crime

### ISSUE THREAD

**Issue by bobyonekenobi86** [friendica/friendica#11773]
Hello,

Could you please recommend a forum / site where is a discussion or description about best practices regarding content moderation?
I want to avoid groups created by members or criminals (child porn, firearms, drugs, etc. etc.).
Are there any automation / bot that filters these?

Are there any tool in the site what members can use like: "report post" or flag somehow the inappropriate content?
Did anyone already implemented an addon like community judges that can delete posts, etc.

Thank you!
Bobyonekenobi86

**Comment by MrPetovan** [comment]
Long story short, there isn't much around Friendica itself.

From the administrator perspective, we have several features regarding moderation:
- Node-wide domain block with optional contact purge
- Node-wide remote contact block with optional content purge
- Local user ban
- Node-wide content deletion

From the user perspective, there are a few more:
- Contact blocking
- Contact ignoring
- Timeline content deletion

But at the moment we do not have a specific content report feature, as a user you can always report content manually by sending a message to your node administrator.

**Comment by tio-trom** [comment]
I think overall for the fediverse an adblock-like solution is needed and the most desirable and opt in if possible. Imagine if there was a list of bots, one of spammers, one of guns, one of whatever. And this list contains people + instances. Created by everyone as a collaboration thing. Now think how useful it would be to enable any of these in friendica and they will be blocked, for you, immediately. 

This way instances can enable the lists by default if so desire, and users can disable them if they desire, or vice-versa.

**Comment by MrPetovan** [comment]
It is fine when a shared blocklist is about websites or domains, but it becomes hairy when it becomes about social media accounts that may be personal. What if regular accounts are put on the list out of spite by someone else? What is the recourse to be removed from the list? How do we even know we're on a list we may not even know about? In the EU, it might even be a GDPR violation if the data used to identify the account contains personal identifiable information (like the full name in the handle and/or the domain).

So far the main collaborative moderation initiative I've seen is regular posts tagged with `#FediBlock` suggesting to block servers or individual accounts. This way the block isn't automated, node admins have to enact the block manually.

**Comment by tio-trom** [comment]
Fair points. However if the accounts are public perhaps it is not a bad approach. But I get your points.

A list with domains that are mostly about scammers, spammers, and bots, can be super useful tho.

**Comment by MrPetovan** [comment]
The list of domains any Friendica node is blocking is available with the reason for each like for mine: https://friendica.mrpetovan.com/friendica

I guess we could add export/import for this list.

**Comment by tio-trom** [comment]
Yes that is a great feature of Friendica! I like it. But would be great to have lists that we can simply import into Friendica for instance.

**Comment by tobiasd** [comment]
> I guess we could add export/import for this list.

This is already possible using the console tool. You can run a daily cron job exporting the list to a CSV file in the webroot of the Friendica node, that can be loaded and imported by other admins.

**Comment by tobiasd** [comment]
> In the EU, it might even be a GDPR 

That is a valid point. You should contact a lawyer for some advice if it was possible to argue that you need to collect these data for the operation of a node to oblige other laws, e.g. the anti harassment laws.

**Comment by bobyonekenobi86** [comment]
[tobiasd](https://github.com/tobiasd) Thanks, I like the cron job idea!

**Comment by tobiasd** [comment]
Just a short update regarding the CLI tool, I wrote a little python helper for this the cauldron to [brewserverblocklist](https://git.friendi.ca/tobias/brewserverblocklist). It collects the list of blocked servers from other Friendica instances and combines them in a semi-automatic way to a CSV file that can then be imported by admins into their systems.
